app/src/app/services/data.service.ts

```diff
@@ -0,0 +1,20 @@
+import {HttpClient} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
```

Comment:
Ohhh boy, reactive js?

Reply:
Ok, so let's chat about this one. Last project I worked on with a reactive core couldn't find people who knew it, ramp up time was really long, and overall, it was very difficult to make large changes without rewriting all the things. In my experience, if you plan on making changes and experimenting reactive slows you down as you have to remodel how your data flows over and over again.

Reply:
yeah Angular uses RxJS. Looking at Observables they're pretty cool.

Reply:
That's not to say we can't use it, I just want to make sure we're making the right decision here and that this addition will meet all our goals

Reply:
@taquitos so i believe we have to use it. I just asked around, and it seems like the only way not to use it is to immediately convert to promises. 

I think the bigger question is should we be using ngRX. That is something that deeply uses React. Observables are really only conceptually minor changes to promises.

Reply:
I think we can punt the ngRX discussion down the line.